src/Controls/src/Xaml/SimplifyOnPlatformVisitor.cs

```diff
@@ -26,60 +54,87 @@ public void Visit(ValueNode node, INode parentNode)
 
 		public void Visit(MarkupNode node, INode parentNode)
 		{
+			//markup was already expanded to element
 		}
 
 		public void Visit(ElementNode node, INode parentNode)
 		{
-			if (node.XmlType.Name != nameof(OnPlatformExtension) || node.XmlType.NamespaceUri != XamlParser.MauiUri)
-				return;
-			if (string.IsNullOrEmpty(TargetFramework))
+			if (Target is null)
 				return;
 
-			string target = null;
-#if NETSTANDARD2_0
-			if (TargetFramework.Contains("-android"))
-#else
-			if (TargetFramework.Contains("-android", StringComparison.Ordinal))
-#endif
-				target = nameof(OnPlatformExtension.Android);
-#if NETSTANDARD2_0
-			if (TargetFramework.Contains("-ios"))
-#else
-			if (TargetFramework.Contains("-ios", StringComparison.Ordinal))
-#endif
-				target = nameof(OnPlatformExtension.iOS);
-#if NETSTANDARD2_0
-			if (TargetFramework.Contains("-macos"))
-#else
-			if (TargetFramework.Contains("-macos", StringComparison.Ordinal))
-#endif
-				target = nameof(OnPlatformExtension.macOS);
-#if NETSTANDARD2_0
-			if (TargetFramework.Contains("-maccatalyst"))
-#else
-			if (TargetFramework.Contains("-maccatalyst", StringComparison.Ordinal))
-#endif
-				target = nameof(OnPlatformExtension.MacCatalyst);
-
-			if (target is null)
-				return;
+			//`{OnPlatform}` markup extension
+			if (node.XmlType.Name == nameof(OnPlatformExtension) && node.XmlType.NamespaceUri == XamlParser.MauiUri)
+			{				
+				if (node.Properties.TryGetValue(new XmlName("", Target), out INode targetNode)
+					|| node.Properties.TryGetValue(new XmlName("", nameof(OnPlatformExtension.Default)), out targetNode))
+				{
+					if (!ApplyPropertiesVisitor.TryGetPropertyName(node, parentNode, out XmlName name))
+						return;
+					if (parentNode is IElementNode parentEnode)
+						parentEnode.Properties[name] = targetNode;
+				}
+				else //no prop for target and no Default set
+				{
+					if (!ApplyPropertiesVisitor.TryGetPropertyName(node, parentNode, out XmlName name))
+						return;
+					//if there's no value for the targetPlatform, ignore the node.
+					//this is slightly different than what OnPlatform does (return default(T))
+					if (parentNode is IElementNode parentEnode)
+						parentEnode.Properties.Remove(name);
+				}
+			}
 
-			if (   node.Properties.TryGetValue(new XmlName("", target), out INode targetNode)
-				|| node.Properties.TryGetValue(new XmlName("", nameof(OnPlatformExtension.Default)), out targetNode))
+			//`<OnPlatform>` elements
+			if (node.XmlType.Name == "OnPlatform" && node.XmlType.NamespaceUri == XamlParser.MauiUri)
 			{
-				if (!ApplyPropertiesVisitor.TryGetPropertyName(node, parentNode, out XmlName name))
+				var onNode = GetOnNode(node, Target) ?? GetDefault(node);
```

Comment:
`onNode` can be null here, right? Is it safe to pass it into `ApplyPropertiesVisitor.TryGetPropertyName`?

Reply:
yes, it's safe, but it'll return false